libraries/rush-sdk/src/proxy.ts

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import type { SpawnSyncReturns } from 'child_process';
+import { JsonFile, JsonObject, Executable } from '@rushstack/node-core-library';
+
+import {
+  tryFindRushJsonLocation,
+  RUSH_LIB_NAME,
+  RushLibModuleType,
+  requireRushLibUnderFolderPath,
+  sdkContext
+} from './helpers';
+
+declare const global: NodeJS.Global &
+  typeof globalThis & {
+    ___rush___rushLibModule?: RushLibModuleType;
+    ___rush___rushLibModuleFromEnvironment?: RushLibModuleType;
+    ___rush___rushLibModuleFromInstallAndRunRush?: RushLibModuleType;
```

Comment:
If we're mapping this globally, are we expressly not supporting the scenario where you need to load multiple versions of rush-lib into the host process to contend with multiple repositories of different versions?

Reply:
Perhaps ultimately all those scenarios will be rewritten to use proxy.ts.  For this PR I was just trying to do the smallest "correctness preserving change"